frontend/src/metabase/admin/settings/components/EmbeddingSettings/hooks.tsx

```diff
@@ -28,3 +34,21 @@ export function useMergeSetting<Key extends SettingKey>(
 
   return mergedSetting;
 }
+
+export const useEmbeddingSetting = <Key extends SettingKey>(
```

Comment:
It looks like there is nothing specific to embedding in this hook, but this is more of a convenience helper for settings in general. The hook name is ok, but I wonder if this pattern might also be useful outside of embedding settings?

Reply:
I think `useSetting` might actually do this for us already, I'll check that

Reply:
Ok, it looks like `useSetting` doesn't do the setting for us, so I can move this code to `metabase/common/hooks` so we can access it in a more general way